src/neo-vm/Types/StackItem.cs

```diff
@@ -21,6 +23,16 @@ public virtual StackItem ConvertTo(StackItemType type)
             throw new InvalidCastException();
         }
 
+        public StackItem DeepCopy()
+        {
+            return DeepCopy(new Dictionary<CompoundType, CompoundType>(ReferenceEqualityComparer.Default));
+        }
+
+        internal virtual StackItem DeepCopy(Dictionary<CompoundType, CompoundType> refMap)
```

Comment:
I think that we can avoid this argument, and make a different logic for maps

Reply:
It's not for map only. All of the `CompoundType` need it.

Reply:
And I find that `Buffer` needs it too. I will fix it tomorrow.